control/controlclient/auto.go

```diff
@@ -407,7 +407,13 @@ func (c *Auto) authRoutine() {
 				c.mu.Unlock()
 
 				c.sendStatus("authRoutine-url", err, url, nil)
-				bo.BackOff(ctx, err)
+				if goal.url == url {
+					// The server sent us the same URL we already tried,
+					// backoff to avoid a busy loop.
+					bo.BackOff(ctx, fmt.Errorf("login URL not changing"))
```

Comment:
Don't do fmt.Errorf, there is no format directive here.

Use errors.New

Reply:
okay